src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -475,6 +475,56 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP087",
                 "Array and object literals are not allowed here.");
+
+            public ErrorDiagnostic UnableToFindPathForModule() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP088",
+                "Unable to find file path for module.");
+
+            public ErrorDiagnostic ErrorOccurredLoadingModule(string failureMessage) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP089",
+                $"An error occurred loading the module. Received failure \"{failureMessage}\".");
+
+            public ErrorDiagnostic ModulePathInterpolationUnsupported() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP090",
+                "String interpolation is unsupported for specifying the module path.");
+
+            public ErrorDiagnostic ModulePathCouldNotBeResolved(string modulePath, string parentPath) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP091",
+                $"Module \"{modulePath}\" could not be resolved relative to \"{parentPath}\".");
+
+            public ErrorDiagnostic CyclicModuleSelfReference() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP092",
+                "This module references its own declaring file, which is not allowed.");
+
+            public ErrorDiagnostic CyclicModule(IEnumerable<string> cycle) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP093",
+                $"The module is involved in a cycle (\"{string.Join("\" -> \"", cycle)}\").");
+
+            public ErrorDiagnostic ExpectedModuleIdentifier() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP094",
+                "Expected a module identifier at this location.");
+
+            public ErrorDiagnostic ExpectedModulePathString() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP095",
+                "Expected a module path string.");
+
+            public ErrorDiagnostic GenericModuleLoadFailure() => new ErrorDiagnostic(

```

Comment:
How likely is this error? (Wondering about actionability...)

Reply:
I spent some time thinking about how to repro this in tests yesterday, and couldn't come up with anything. It's just there to handle the case where we haven't been able to load the module, but we didn't store failure information (aka just handling a null value which I don't expect to see). I'll see if I can rework this logic to avoid the potentiality of null.

Reply:
If we can inject a bad mock implementation, that could be a way to test it. (That's how I'm testing the fatal exception handling in the language server.)

Reply:
I've removed the error entirely